src/v/rpc/connection_cache.h

```diff
@@ -94,17 +94,22 @@ class connection_cache final
                   return ss::futurize<ret_t>::convert(
                     rpc::make_error_code(errc::missing_node_rpc_client));
               }
-              return cache.get(node_id)
-                ->get_connected(connection_timeout.timeout_at())
-                .then([f = std::forward<Func>(f)](
-                        result<ss::lw_shared_ptr<rpc::transport>>
-                          transport) mutable {
-                    if (!transport) {
-                        // Connection error
-                        return ss::futurize<ret_t>::convert(transport.error());
-                    }
-                    return ss::futurize<ret_t>::convert(
-                      f(Protocol(transport.value())));
+              return ss::do_with(
+                cache.get(node_id),
+                [connection_timeout = connection_timeout.timeout_at(),
+                 f = std::forward<Func>(f)](auto& transport_ptr) mutable {
+                    return transport_ptr->get_connected(connection_timeout)
+                      .then([f = std::forward<Func>(f)](
```

Comment:
Just curious if you tried coroutinizing this as some unsharded template method, and if you did but opted for this what the snag was? the inferred signature? the `convert()` calls?

Reply:
I didn't try coro-ing this, just playing safe, didn't want to introduce new bugs.